### PR TITLE
Ft login functionality #184359072

### PR DESCRIPTION
**Pull request by fabrice-rw**

### CODE REVIEW

**Review by jeandamas — DISMISSED** [review]

**Comment by key-joshua** [comment]
Check some of your tests are not failing to pass and also resolve conflicts

**Comment by key-joshua** [comment]
Check some of your tests are failing to pass and also resolve conflicts

**Comment by key-joshua** [comment]
Check some of your tests are failing to pass and also resolve conflicts as well.

**Comment by key-joshua** [comment]
Check some of checks are failing to pass and also resolve conflicts

**Review by TresorRw — APPROVED** [review]

File: src/routes/authroutes.ts [hunk 1]
```diff
@@ -18,31 +19,33 @@ const router = Router()
...
  * */
 router.post('/signup', signupValidation, auth.signup)
+router.post('/login', loginValidation, auth.Login)
 router.get('/users', auth.getAlluser)
 router.delete('/delete/:id', auth.deleteUser)
 router.get('/sendcode/:phone', auth.sendCode)
 router.get('/verify/:phone/:code', auth.verify2FA)
-router.get('/logout', auth.logout)
-/* this delete user route is not protected it is just for testing and setting up the project*/
+// router.post('/logout', auth.logout)
```
Comment (EmyRukundo) [inline src/routes/authroutes.ts:47]: any reason for this ?

File: src/controllers/authController.ts [hunk 2]
```diff
@@ -109,11 +118,49 @@ class auth {
...
+      const findUser = await USER.findOne({
+        where: { email: email },
+        attributes: ['id', 'firstName', 'lastName', 'email', 'role', 'password'],
+      })
+      if (!findUser) {
+        res.status(404).json({ message: "User not found" })
+      } else {
+        const dbPassword = findUser.dataValues.password
+        const decreptedPassword = await bcrypt.compare(password, dbPassword)
+        // console.log(decreptedPassword)
```
Comment (EmyRukundo) [inline src/controllers/authController.ts:135]: remove this

File: src/controllers/authController.ts [hunk 3]
```diff
@@ -63,13 +72,13 @@ class auth {
       })
     }
   }
-    static async signup(req: Request, res: Response) {
-        try {
-        // await USER.drop()
-      const {firstName, lastName, email, role, password} = req.body
+  static async signup(req: Request, res: Response) {
+    try {
+      // await USER.drop()
```
Comment (EmyRukundo) [inline src/controllers/authController.ts:77]: you can remove this

File: src/controllers/authController.ts [hunk 4]
```diff
@@ -1,10 +1,20 @@
 import USER from '../models/User'
```
Comment (EmyRukundo) [inline src/controllers/authController.ts:1]: Hi Fabrice, I can not see the test codes >

**Review by EmyRukundo — CHANGES_REQUESTED** [review]

**Review by Rukundojoseph — APPROVED** [review]

**Review by EmyRukundo — APPROVED** [review]

**Pull request merged**